tools/clang/unittests/HLSLExec/ExecutionTest.cpp

```diff
@@ -13236,6 +13205,173 @@ void ExecutionTest::WaveSizeTest() {
   }
 }
 
+bool TestShaderRangeAgainstRequirements(UINT shaderminws, UINT shadermaxws,
+                                        UINT minws, UINT maxws) {
+  if (shaderminws > maxws) {
+    return false;
+  }
+  if (shadermaxws < minws) {
+    return false;
+  }
+  return true;
+}
+
+void RunWaveSizeRangeTest(UINT minWaveSize, UINT maxWaveSize,
+                          std::shared_ptr<st::ShaderOpSet> ShaderOpSet,
+                          CComPtr<ID3D12Device> pDevice,
+                          dxc::DxcDllSupport &m_support) {
+  // format shader source
+  const char waveSizeTestShader[] =
+      "struct TestData { \r\n"
+      "  uint count; \r\n"
+      "}; \r\n"
+      "RWStructuredBuffer<TestData> data : register(u0); \r\n"
+      "\r\n"
+      "// Note: The 3 WAVESIZE variables below will be defined via compiler "
+      "option -D\r\n"
+      "[wavesize(MINWAVESIZE, MAXWAVESIZE, PREFWAVESIZE)]\r\n"
+      "[numthreads(" stringify(MAX_WAVESIZE) "*2,1,1)]\r\n"
+                                             "void main(uint3 tid : "
+                                             "SV_DispatchThreadID ) { \r\n"
+                                             "  data[tid.x].count = "
+                                             "WaveGetLaneCount(); \r\n"
+                                             "}\r\n";
+
+  struct WaveSizeTestData {
+    uint32_t count;
+  };
+
+  for (UINT minShaderWaveSize = 4; minShaderWaveSize <= maxWaveSize;
+       minShaderWaveSize *= 2) {
+    for (UINT maxShaderWaveSize = minShaderWaveSize * 2;
+         maxShaderWaveSize <= 128; maxShaderWaveSize *= 2) {
+      for (UINT prefShaderWaveSize = minShaderWaveSize;
+           prefShaderWaveSize <= maxShaderWaveSize; prefShaderWaveSize *= 2) {
+
+        // For now, only allow valid shader wave ranges
+        // TODO: Create callback function for CreateComputePipelineState to
+        // handle expected errors.
```

Comment:
A test for the runtime API validation should be separate from this test, so we should simply skip any incompatible sizes without needing to comment about them in the log.